src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php

```diff
@@ -114,6 +142,8 @@ protected function doGetArguments(Request $request, $controller, array $paramete
                 $arguments[] = $request;
             } elseif ($param->isDefaultValueAvailable()) {
                 $arguments[] = $param->getDefaultValue();
+            } elseif ($this->supportsParameterType && null !== ($type = $param->getType()) && $type->allowsNull()) {
```

Comment:
Ah, yes I was thinking you were still referring to the actual method in the type. I'm running the tests locally and it seems to work fine